pkg/operator/controller/metrics/metrics.go

```diff
@@ -0,0 +1,46 @@
+package metrics
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	// DNSesTotal is a Prometheus gauge metric which holds the total number
+	// of DNSes.
+	DNSesTotal = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "dns_operator_dnses_total",
+		Help: "Total number of DNSes",
+	}, []string{"dnses"})
+
+	// DNSesUsingForwardingAPI is a Prometheus gauge metric which indicates
+	// how many DNS operands are using the Forwarding API.
+	DNSesUsingForwardingAPI = prometheus.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
I'd just as soon use a Boolean, but Prometheus does not appear to have a Boolean metric type.  My thinking is that we will add an additional metric for each major feature so as to provide insight as to whether that feature is being used; would it make more sense to have a single "features" metric with a label for each feature, or is there some more idiomatic approach?

Reply:
@smarterclayton can you give us a little guidance on the idioms for metric labelling here?